provider/src/test/kotlin/org/gradle/kotlin/dsl/resolver/SourcePathProviderTest.kt

```diff
@@ -34,11 +32,43 @@ class SourcePathProviderTest : FolderBasedTest() {
             sourcePathFor(
                 classPath = ClassPath.EMPTY,
                 projectDir = folder("project"),
-                gradleHomeDir = folder("gradle")).asFiles,
+                gradleHomeDir = folder("gradle"),
+                sourceDistributionResolver = mock(SourceDistributionResolver::class.java)).asFiles,
             hasItems(
                 folder("project/buildSrc/src/main/foo"),
                 folder("project/buildSrc/src/main/bar"),
                 folder("gradle/src/gradle-foo"),
                 folder("gradle/src/gradle-bar")))
     }
+
+    @Test
+    fun `when src dir is missing from Gradle distribution, it will try to download it`() {
+        withFolders {
+            "project" {
+                "buildSrc/src/main" {
+                    +"foo"
+                    +"bar"
+                }
+            }
+            "gradle" {
+
+            }
+        }
+
+        val resolver = mock(SourceDistributionResolver::class.java)
```

Comment:
:x: Please check out [this test](https://github.com/gradle/kotlin-dsl/blob/21eba6dc6e8d3a672c530e680481a1b367ef031f/provider/src/test/kotlin/org/gradle/kotlin/dsl/DependencyHandlerExtensionsTest.kt#L122) for inspiration on setting up this mock.